utils/s3/aws_error.cc

```diff
@@ -75,6 +75,35 @@ aws_error aws_error::parse(seastar::sstring&& body) {
     return ret_val;
 }
 
+aws_error aws_error::from_http_code(seastar::http::reply::status_type http_code) {
+    const auto& all_errors = get_errors();
+    switch (http_code) {
+    case seastar::http::reply::status_type::unauthorized:
+    case seastar::http::reply::status_type::forbidden:
+        return all_errors.at("AccessDenied");
+    case seastar::http::reply::status_type::not_found:
+        return all_errors.at("ResourceNotFound");
+    case seastar::http::reply::status_type::too_many_requests:
+        return all_errors.at("SlowDown");
+    case seastar::http::reply::status_type::internal_server_error:
+        return all_errors.at("InternalError");
+    case seastar::http::reply::status_type::bandwidth_limit_exceeded:
+        return all_errors.at("Throttling");
+    case seastar::http::reply::status_type::service_unavailable:
+        return all_errors.at("ServiceUnavailable");
+    case seastar::http::reply::status_type::request_timeout:
+    case seastar::http::reply::status_type::page_expired:
+    case seastar::http::reply::status_type::login_timeout:
+    case seastar::http::reply::status_type::gateway_timeout:
+    case seastar::http::reply::status_type::network_connect_timeout:
+    case seastar::http::reply::status_type::network_read_timeout:
+        return all_errors.at("RequestTimeout");
```

Comment:
Not sure if it's a good way to propagate back the retryable::yes|no bit :( I would say it's better to introduce two (or more if required) aws_error_type-s above `SERVICE_EXTENSION_START_RANGE` and just return the corresponding aws_error-s here

Reply:
added new errors for each of these